packages/api/src/routes/medical/internal-patient.ts

```diff
@@ -53,27 +56,35 @@ router.post(
     let failedUpdateCount = 0;
     for (const facility of facilities) {
       const patients = await getPatients({ cxId, facilityId: facility.id });
-      const patientUpdates = [];
-      for (const patient of patients) {
-        const patientUpdate: PatientUpdateCmd = {
-          id: patient.id,
-          cxId: patient.cxId,
-          address: patient.data.address,
-          dob: patient.data.dob,
-          firstName: patient.data.firstName,
-          genderAtBirth: patient.data.genderAtBirth,
-          lastName: patient.data.lastName,
-          contact: patient.data.contact,
-          personalIdentifiers: patient.data.personalIdentifiers,
-        };
-        patientUpdates.push(updateInFHIRAndCW(patientUpdate, facility.id));
-      }
-      const result = await Promise.allSettled(patientUpdates);
-      for (const patientUpdate of result) {
-        if (patientUpdate.status === "rejected") {
-          failedUpdateCount++;
+
+      const executeInSequence = async (patients: Patient[]) => {
+        for (const patient of patients) {
+          const patientUpdate: PatientUpdateCmd = {
+            id: patient.id,
+            cxId: patient.cxId,
+            address: patient.data.address,
+            dob: patient.data.dob,
+            firstName: patient.data.firstName,
+            genderAtBirth: patient.data.genderAtBirth,
+            lastName: patient.data.lastName,
+            contact: patient.data.contact,
+            personalIdentifiers: patient.data.personalIdentifiers,
+          };
+          try {
+            await updateInFHIRAndCW(patientUpdate, facility.id);
+          } catch (error) {
+            console.log(`Failed to update patient ${patient.id} - ${errorToString(error)}`);
+            failedUpdateCount++;
+          }
         }
-      }
+      };
+      await executeAsynchronously(
+        patients,
+        async patients => {
+          return executeInSequence(patients);
+        },
+        { numberOfAsyncRuns: 10 }
+      );
```

Comment:
The previous code was hammering down our server/db when 1,000+ patients in one facility - e.g.: https://metriport.slack.com/archives/C04GEQ1GH9D/p1697240413099569?thread_ts=1697163054.417309&cid=C04GEQ1GH9D